Show notifications for replied messages and handle oreo quick reply actions

spuday90 opened the pull request:
closes #189

spuday90 commented:
Hi @adeekshith Please check on your device, on mine somehow getting 2 replies.

adeekshith commented on app/src/main/java/com/parishod/watomatic/NotificationService.java:
```diff
@@ -88,6 +94,42 @@ private void sendReply(StatusBarNotification sbn) {
+                && remoteInputs.size() == 0 ) {
+                for(int x = 0; x < act.getRemoteInputs().length; x++) {
+                    RemoteInput remoteInput = act.getRemoteInputs()[x];
+                    if((remoteInput.getResultKey().toLowerCase().contains("reply")
```
Using English word "reply" may not work for other languages

spuday90 replied:
Well actually i realised we dont need to check "reply" as remoteInput object only will for reply

adeekshith reviewed (changes requested):
Commented items and multiple replies needs to be fixed

adeekshith commented:
@spuday90 
- Can we separate debug logs sharing code? It has external storage permissions so we might want to avoid that for now.
- What issues are we having with constrained layout?

spuday90 commented:
> @spuday90
> 
>     * Can we separate debug logs sharing code (cherry-pick)? It has external storage permissions so we might want to avoid that for now.
> 
>     * What issues are we having with constrained layout?

In the latest commits we have removed external storage permission and moved the logs write to internal storage
With constraint layout spacing was coming between main fragment content and dismiss notification settings, i think we can use biasing to fix that but felt linear layout is better and faster in that case so changed it

adeekshith commented:
Also keeping the notification after reply only seems to be working for Whatsapp

spuday90 commented:
> Also keeping the notification after reply only seems to be working for Whatsapp

Oh!, have to debug which reply intent it is taking

adeekshith commented:
> > Also keeping the notification after reply only seems to be working for Whatsapp
> 
> Oh!, have to debug which reply intent it is taking
> 

I think manually replying to facebook via notification also dismisses it so not really app's fault

adeekshith commented:
@spuday90 conflicts

adeekshith commented:
@spuday90 strings.xml and arrays.xml needs to be checked too. they replaced strings in main branch if not conflicting

adeekshith reviewed (changes requested):
- Needs to be updated with latest "main" (merge/rebase)
- Resolve conflicts with main
- Fix strings.xml and arrays.xml if not already done

- Remove logging to file code

- Notification Setting
  - Remove "Oreo+ only" if it works on previous ones also
  - Move setting to "Settings activity"
  - Change switch label to "Dismiss replied notifications"
  - Remove help tip below the switch ("Turn On to dismiss...")

adeekshith commented:
I see this in the latest branch with Watomatic attribution enabled

![Screenshot_20210405-173150_WhatsApp.jpg](https://user-images.githubusercontent.com/2568945/113634653-b390a900-9634-11eb-86a4-a2112387e2dc.jpg)

adeekshith commented:
Also clicking on Notification button is behaving like "Share" button. Is this happening only on mine?
Tried deleting local branch and pulled latest.

https://user-images.githubusercontent.com/2568945/113635640-9e1c7e80-9636-11eb-8c54-d872ad623718.mp4

adeekshith commented:
- Watomatic notification icon too small. Let me know if it needs anything done from from me end.
- Can we make it normal ("Alert" or similar) notification instead of Silent notification?
- WhatsApp original notification is still present. Should we dismiss that?

<img width="298" alt="Screen Shot 2021-04-05 at 5 48 22 PM" src="https://user-images.githubusercontent.com/2568945/113635953-331f7780-9637-11eb-8ed1-217017703c3c.png">

spuday90 commented:
> Also clicking on Notification button is behaving like "Share" button. Is this happening only on mine?
> Tried deleting local branch and pulled latest.
> notification-btn-share.mp4

I completely removed share code in latest commit, so it should have crashed instead
If something was missed while removing that

spuday90 commented:
>     * Watomatic notification icon too small. Let me know if it needs anything done from from me end.
> 
>     * Can we make it normal ("Alert" or similar) notification instead of Silent notification?
> 
>     * WhatsApp original notification is still present. Should we dismiss that?
> 
> 
> <img alt="Screen Shot 2021-04-05 at 5 48 22 PM" width="298" src="https://user-images.githubusercontent.com/2568945/113635953-331f7780-9637-11eb-8ed1-217017703c3c.png">

1. For notification icon I have used the foreground one for now, we should generate actual notification icon, or we can use app icon, App icon will look fine in status bar but when status bar pulled down then icon will look greyed out.
2.Yes we can change silent notification to standard notification
3.Yes we should dismiss whatsapp notification, as for other apps it is anyhow dismissing so itll be consistent

adeekshith commented:
Have you seen this happening?
I am getting multiple replies on this branch
![Screenshot_20210406-091641_WhatsApp.jpg](https://user-images.githubusercontent.com/2568945/113725182-b1702e00-96b8-11eb-9cf8-4f8761c2ad6e.jpg)

adeekshith commented on app/src/main/res/values/strings.xml:
```diff
@@ -55,8 +55,17 @@
     <string name="watomatic_wato_message_url" translatable="false">https://www.reddit.com/r/watomatic/comments/lrkn7j/fellow_watos_share_your_creative_wato_message/</string>
 
+    <!-- Share debug logs -->
+    <string name="app_logs_file_name">WatomaticLogs.txt</string>
```
@spuday90 Mark all share debug logs as `translatable="false"`

adeekshith commented on app/src/main/res/values/strings.xml:
```diff
@@ -55,8 +55,17 @@
+
     <!-- keys and app internal strings -->
     <string name="key_pref_app_language" translatable="false">pref_app_language</string>
+    <string name="pref_show_notification" translatable="false">pref_show_notification</string>
```
`pref_show_notification` -> `pref_show_notification_replied_msg` for value to future proof (but optional)

adeekshith commented:
@spuday90 Approve this but I am still seeing these:
- Notifications are silent
- Clicking on notification is not opening any app

I remember you fixing it before but not sure. If that is not fixed, we can merge this PR and file an issue.
